app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt

```diff
@@ -151,7 +153,21 @@ class LaunchActivity : AppCompatActivity() {
                     level = SentryLevel.INFO
                 })
 
-                setDeepLinkFileExtra(it.destinationDriveId, it.destinationRemoteFolderId)
+                CoroutineScope(Dispatchers.IO).launch {
```

Comment:
Never create a `CoroutineScope` that is not strongly referenced anywhere.
If the underlying coroutine is running on code that doesn't keep a strong reference to the cancellable continuation (and therefore the `Job` that is connected to parents and holds everything together), the coroutine could silently get garbage collected.

Here, you are in a `ComponentActivity` subclass, so you can use `lifecycleScope`.

To avoid unneeded blocking code, you could do something like this:

```kotlin
lifecycleScope.launch {
    val allUserIds = UserDatabase().userDao().allUsers.first().map { it.id }
    // Rest of the code, granted it's not doing blocking I/O
}
```